zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,288 @@
+from datetime import datetime
+
+import flask
+from flask import request
+
+from .utils.json_result import json_result
+from .utils.route_wrappers import cross_domain, with_session
+from . import api
+import zeeguu
+from zeeguu.model import User, Cohort, UserActivityData, Session, Language
+import sqlalchemy
+from flask import jsonify
+import json
+import random
+import datetime
+
+# class function wrapper
+def class_function_checker(class_id):
+    from zeeguu.model import TeacherCohortMap
+    link = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+    for l in link:
+        if l.user_id == flask.g.user.id:
+            return True
+    return False
+
+
+# Takes user_id and returns user.name that corresponds
+#@api.route("/get_user_name/<id>", methods=["GET"])
+@with_session
+def get_user_name(id):
+    user = User.query.filter_by(id=id).one()
+    return user.name
+
+
+# Checks to see if user has a valid active session
+@api.route("/has_session", methods=["GET"])
```

Comment:
this is not used anywhere. 
why is it needed? 

seems to replicate the behavior in @with_session which can be triggered by the /validate endpoint (in zeeguu_api/api/sessions)

Reply:
but can't you use /validate instead?